aten/src/ATen/native/transformers/cuda/attention.cu

```diff
@@ -376,6 +376,7 @@ __host__ std::tuple<Tensor, Tensor, Tensor> transform_bias_rescale_qkv_cuda(
         }
         if (qkv.is_nested()) {
           auto* nt_qkv = get_nested_tensor_impl(qkv);
+          const at::Tensor& nt_qkv_buffer = nt_qkv->get_buffer();
```

Comment:
@swolchok The goal of this PR is to sub out the buffer_ that is stored on NestedTensor with instead a storage. Calling `packed_accessor64` in theory should be possible for NestedTensor but there is some size checks that currently fail. `get_buffer` now returns a regular `at::Tensor` instead of a `const at::Tensor&`. This tensor gets gets constructed at function execution from the underlying storage of the NestedTensorImpl. The change here seems a little like a compiler hack since the version I was trying to call gets deleted: https://github.com/pytorch/pytorch/blob/1164c83c3c3070c82618350d57c7bf65ec56f0f1/aten/src/ATen/core/TensorBase.h#L562

It seems to work but not sure if I am somehow foot gunning here.

Reply:
this is fine, but note that it's the same as if you had written (more clearly IMO) `at::Tensor nt_qkv_buffer = nt_qkv->get_buffer();` . It builds because of temporary lifetime extension -- https://abseil.io/tips/107